### PR TITLE
feat: create named producers

### LINKED ISSUE #30: feat: Implement named producers
## Issue Report

Allow the producer configuration based on a name and create an accessor to get the producer

## Expected Behavior

Support named and typed configuration

## Actual Behavior

Today we only support producers based on a type to bind its configuration

### PR DESCRIPTION
**Pull request by filipeesch**
Closes #30

### CODE REVIEW

File: src/KafkaFlow/Configuration/KafkaFlowConfigurator.cs [hunk 1]
```diff
@@ -33,11 +37,19 @@ public IKafkaBus CreateBus(IDependencyResolver resolver)
         {
             var scope = resolver.CreateScope();
 
+            foreach (var producerConfig in this.configuration.Clusters.SelectMany(x => x.Producers))
+            {
+                ProducerManager.Instance.AddOrUpdate(
+                    new MessageProducer(
```
[the remark this answers is not in the document]
Comment (ailtonguitar) [reply src/KafkaFlow/Configuration/KafkaFlowConfigurator.cs:43]: With this approach, you will create duplicated instances of MessageProducer for each producerconfig.

File: src/KafkaFlow/Producers/ProducerManager.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        private readonly ConcurrentDictionary<string, IMessageProducer> producers =
+            new ConcurrentDictionary<string, IMessageProducer>();
+
+        private ProducerManager()
+        {
+        }
+
+        public static readonly ProducerManager Instance = new ProducerManager();
+
+        public IMessageProducer GetProducer(string name) =>
```
[the remark this answers is not in the document]
Comment (filipeesch) [reply src/KafkaFlow/Producers/ProducerManager.cs:17]: Added

File: src/KafkaFlow/Configuration/ProducerConfigurationBuilder.cs [hunk 3]
```diff
@@ -51,17 +59,20 @@ public ProducerConfiguration Build(ClusterConfiguration clusterConfiguration)
...
 
-            this.DependencyConfigurator.AddSingleton(
-                typeof(IMessageProducer<>).MakeGenericType(this.producerType),
-                resolver => Activator.CreateInstance(
-                    typeof(MessageProducer<>).MakeGenericType(this.producerType),
-                    resolver,
-                    configuration));
+            if (this.producerType != null)
+            {
+                this.DependencyConfigurator.AddSingleton(
```
[the remark this answers is not in the document]
Comment (filipeesch) [reply src/KafkaFlow/Configuration/ProducerConfigurationBuilder.cs]: I simplified this registering the producer in the AddProducer method

File: src/KafkaFlow/Configuration/KafkaFlowConfigurator.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (filipeesch) [reply src/KafkaFlow/Configuration/KafkaFlowConfigurator.cs]: You are right. I moved this registration to KafkaBus.

**Comment by ailtonguitar** [comment]
lgtm

**Comment by dougolima** [comment]
Please, add an example of usage on samples project :)

**Pull request merged**